### ISSUE TITLE
WpMailCatcher\Loggers\Logger::getAttachmentLocations() expects an array of attachments although wp_mail() allows a string

### ISSUE THREAD

**Issue by pbiron** [JWardee/wp-mail-catcher#48]
I just noticed the following in my PHP error log:

```
[17-Apr-2019 16:55:44 UTC] PHP Warning:  array_walk() expects parameter 1 to be array, string given in path\to\wp-mail-catcher\src\Loggers\Logger.php on line 111
```

I'm writing a plugin that at one point sends an email with an attachment and I'm calling `wp_mail()` as follows:

```
$to = get_option( 'report-email-to' );
$subject = get_option( 'report-email-subject' );
$message = get_option( 'report-email-message' );
$report_file = '/tmp/report.csv';

// send the email.
wp_mail( $to, $subject, $message , $headers, $report_file );
```

which is acceptable according to [wp_mail()](https://developer.wordpress.org/reference/functions/wp_mail/https://developer.wordpress.org/reference/functions/wp_mail/).

So, you can either add

```
if ( ! is_string($attachments)) {
	$attachments = array($attachments);
}
```
to `getAttachmentLocations()` before the call to `array_walk()`

or change the `array_walk()` call to:

```
array_walk((array) $attachments, function(&$value) {
	$value = str_replace(GeneralHelper::$uploadsFolderInfo['basedir'] . '/', '', $value);
});
```

depending on your preferred coding style (I think the second is simpler, the first is more explicit).

**Comment by pbiron** [comment]
I realized that the second suggestion above doesn't actually work, because `array_walk()` expects the first param to be by reference...and PHP can't pass a cast variable by reference.

However, a simpler form of the first suggestion does work:

```
$attachments = (array) $attachments;
```

**Comment by JWardee** [comment]
Fixed in #49, please reopen if this is still an issue
